script.js

```diff
@@ -1,4 +1,668 @@
+    }
+    // Once, correct (ace) card is selected, change player's ace card based on input and update player's points
+    if (input == ACE_ELEVEN || input == ACE_ONE) {
+      myOutputValue = changeAceValue(input);
+    }
+    // Allow player to draw extra card and update player's points
+    if (input == INPUT_HIT) {
+      myOutputValue = playerDrawsExtraCard();
+    }
+    // Allow player to stand and switch to next mode
+    if (input == INPUT_STAND) {
+      // Ensure that player meets minimum game score required, else change game mode back
+      if (playerPoints < 17) {
+        gameMode = PLAYER_ACTION;
+
+        return `${
+          userName[currentPlayer]
+        }, you need at least 17 points. <br><br>Please type 'hit' to draw another card. <br><br> ${generatePlayerCardList()}`;
+      }
+      gameMode = COMPUTER_CALCULATES;
+    }
+  }
+  if (gameMode == CHANGE_ACE_MODE) {
+    // Ensure player has selected an ace card
+    myOutputValue = checkForAce(input);
+  }
+  if (gameMode == COMPUTER_CALCULATES) {
+    // Check if computer has enough cards and add if necessary
+    myOutputValue = calculateComputerScore();
+  }
+  if (gameMode == FINAL_RESULT) {
+    // Compare player and computer cards and determine winner
+    myOutputValue = determineFinalResult();
+    if (currentPlayer == numPlayers) {
+      gameMode = GAME_SUMMARY;
+    } else {
+      gameMode = SELECT_PLAYER;
+    }
+  }
+  // Allow player to reset entire game and reselect number of players and usernames
+  if (input == INPUT_RESET) {
+    gameMode = WAITING_FOR_NAME;
+    myOutputValue =
+      myOutputValue = `Hello player(s)! <br><br> Please enter the number of people playing`;
+  }
+  // Output final results of all players
+  if (gameMode == GAME_SUMMARY) {
+    myOutputValue = displayGameSummary();
+    gameMode = ASK_FOR_BET;
+  }
   return myOutputValue;
 };
+
+var inputDisplay = function () {
+  //  Change input message based on mode
+  if (gameMode == WAITING_FOR_NUM_PLAYERS) {
+    return "WELCOME TO SHANNON'S CASINO! <br> <br>Click submit to start playing!:";
+  } else if (gameMode == WAITING_FOR_NAME) {
+    return "How many of you are playing today?:";
+  } else if (gameMode == INSTRUCTIONS) {
+    return `Wow! ${numPlayers} players! <br><br> What are your names?:`;
+  } else if (gameMode == ASK_FOR_BET) {
+    return "The game is about to begin!: ";
+  } else if (gameMode == TAKE_BET) {
+    return "Place your bets!:";
+  } else if (gameMode == DEAL_CARDS) {
+    return "Feeling rich huh?:";
+  } else if (gameMode == SELECT_PLAYER) {
+    return "What would you like to do?:";
+  } else if (gameMode == PLAYER_ACTION) {
+    return "What would you like to do?:";
+  } else if (gameMode == CHANGE_ACE_MODE) {
+    return "Hmmm... Let me check if you have an ace:";
+  } else if (gameMode == FINAL_RESULT) {
+    return "Tabulating results!:";
+  } else if (gameMode == GAME_SUMMARY) {
+    return "The final results are!:";
+  }
+};
```

Comment:
great job creating a function dynamically render the text seen on the screen. However, since you have around 11 cases here, using if and else if statements are not very ideal. An alternative to if and else if statements would be to use switch statements that are designed specifically for cases where you have numerous cases. Using a switch would help to make this look more readable. Here's a link for switch statement: https://www.w3schools.com/js/js_switch.asp